scripts/generator/graphql_generator/csharp/SDK/ResponseMergeUtil.cs.erb

```diff
@@ -0,0 +1,69 @@
+namespace <%= namespace %>.SDK {
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    public class ResponseMergeUtil {
+        public delegate void MergeFieldDelegate(string field, IDictionary into, IDictionary responseA, IDictionary responseB);
+
+        public static void DoNotMergeIfExists(string field, IDictionary into, IDictionary responseA, IDictionary responseB) {
+            if (!responseA.Contains(field)) {
+                into[field] = responseB[field];
+            }
+        }
+
+        public static void MergeOverwrite(string field, IDictionary into, IDictionary responseA, IDictionary responseB) {
+            into[field] = responseB[field];
+        }
+
+        public static void MergeListsConcat(string field, IDictionary into, IDictionary responseA, IDictionary responseB) {
+            List<object> newList = new List<object>();
+            IList listA = (IList) responseA[field];
+            IList listB = (IList) responseB[field];
+
+            foreach(object item in listA) {
+                newList.Add(item);
+            }
+
+            foreach(object item in listB) {
+                newList.Add(item);
+            }
+
+            into[field] = newList;
+        }
+
+        private Dictionary<string, MergeFieldDelegate> FieldMergers = new Dictionary<string, MergeFieldDelegate>();
+        private Dictionary<string, ResponseMergeUtil> ObjectMergers = new Dictionary<string, ResponseMergeUtil>();
+
+        public void AddFieldMerger(string field, MergeFieldDelegate merger) {
+            FieldMergers[field] = merger;
+        }
+
+        public void AddObjectMerger(string field, ResponseMergeUtil merger) {
+            ObjectMergers[field] = merger;
+        }
+
+        public Dictionary<string, object> Merge(IDictionary responseA, IDictionary responseB) {
+            Dictionary<string, object> MergedResponse = new Dictionary<string, object>();
+
+            foreach(string field in responseA.Keys) {
```

Comment:
We want to clone the list instead of assign by reference. Just trying to ensure that no mutations happen down the line to the originally queried data.